pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/periodictask/ControllerPeriodicTask.java

```diff
@@ -103,10 +105,16 @@ public final ControllerMetrics getControllerMetrics() {
    * <p>
    * Override one of this method, {@link #processTable(String)} or {@link #processTable(String, C)}.
    */
-  protected void processTables(List<String> tableNamesWithType) {
+  protected void processTables(List<String> tableNamesWithType, String taskParamsJson) {
     int numTables = tableNamesWithType.size();
     LOGGER.info("Processing {} tables in task: {}", numTables, _taskName);
-    C context = preprocess();
+    C context;
+    try {
+      context = preprocess(taskParamsJson);
+    } catch (IOException e) {
```

Comment:
We shouldn't need to catch this exception when the key-values are already put into the properties

Reply:
Moving to Properties instead of json makes `preprocess` not throw exceptions. Ack